cmd/aries-agent-mobile/pkg/api/storage.go

```diff
@@ -6,64 +6,95 @@ SPDX-License-Identifier: Apache-2.0
 
 package api
 
-// StoreIterator is the iterator for the latest snapshot of the underlying store.
-type StoreIterator interface {
-	// Next moves the iterator to the next key/value pair.
-	// It returns false if the iterator is exhausted.
-	Next() bool
-
-	// Free releases associated resources. Release should always success
-	// and can be called multiple times without causing error.
-	Free()
-
-	// Error returns any accumulated error. Exhausting all the key/value pairs
-	// is not considered to be an error.
-	Error() error
-
-	// Key returns the key of the current key/value pair, or nil if done.
-	// The caller should not modify the contents of the returned slice, and
-	// its contents may change on the next call to any 'seeks method'.
-	Key() []byte
-
-	// Value returns the value of the current key/value pair, or nil if done.
-	// The caller should not modify the contents of the returned slice, and
-	// its contents may change on the next call to any 'seeks method'.
-	Value() []byte
+import spi "github.com/hyperledger/aries-framework-go/spi/storage"
+
+// StorageProvider represents a storage provider.
+type StorageProvider interface {
+	// OpenStore opens a store with the given name and returns a handle.
+	// If the store has never been opened before, then it is created.
+	// Store names are not case-sensitive. If name is blank, then an error will be returned.
+	OpenStore(name string) (Store, error)
+
+	// SetStoreConfig sets the configuration on a store.
+	// The store must be created prior to calling this method.
+	// If the store cannot be found, then an error wrapping ErrStoreNotFound will be returned.
+	// If name is blank, then an error will be returned.
+	SetStoreConfig(name string, config spi.StoreConfiguration) error
+
+	// GetStoreConfig gets the current store configuration.
+	// The store must be created prior to calling this method.
+	// If the store cannot be found, then an error wrapping ErrStoreNotFound will be returned.
+	// If name is blank, then an error will be returned.
+	GetStoreConfig(name string) (spi.StoreConfiguration, error)
+
+	// GetOpenStores returns all currently open stores.
+	GetOpenStores() []Store
+
+	// Close closes all stores created under this store provider.
+	// For persistent store implementations, this does not delete any data in the stores.
+	Close() error
 }
 
-// Store is the storage interface.
+// Store represents a storage database.
```

Comment:
Isn't this defined in spi, imported above?

Reply:
My understanding from talking to @soluchok was that this duplication of interfaces is necessary for the mobile bindings. Perhaps @soluchok can elaborate?

Reply:
When you determine if it's needed, it might be an idea to add a comment to this file for future readers.

Reply:
@soluchok, If you let me know the reasoning I can add a comment

Reply:
@troyronda After talking to @soluchok, I now better understand the reason we need these additional interfaces. I've added comments to the files explaining it for future readers.